aten/src/ATen/native/Loss.cpp

```diff
@@ -390,20 +390,18 @@ Tensor& mse_loss_backward_out(Tensor& grad_input, const Tensor& grad_output,
 }
 
 Tensor l1_loss(const Tensor& input, const Tensor& target, int64_t reduction) {
-  auto loss = input.sub(target).abs_();
+  auto diff = input.sub(target);
+  auto loss = diff.is_complex() ? diff.abs() : diff.abs_();
   return apply_loss_reduction(loss, reduction);
 }
 
-Tensor& l1_loss_out(Tensor&result, const Tensor& input, const Tensor& target, int64_t reduction) {
-  if (reduction != Reduction::None) {
-    auto loss = input.sub(target).abs_();
-    if (reduction == Reduction::Mean) {
-      at::mean_out(result, loss, 0);
-    } else {
-      at::sum_out(result, loss, 0);
-    }
-  } else {
-    at::sub_out(result, input, target).abs_();
+Tensor& l1_loss_out(Tensor& result, const Tensor& input, const Tensor& target, int64_t reduction) {
+  auto diff = at::sub_out(result, input, target);
```

Comment:
does this cause warnings?  Because we usually warn when the result is resized (not from something 0-sized).

Reply:
Yep

Reply:
Should be fixed in the latest update. When the shape of `result` matches the post-reduce shape a warning should no longer appear.

Reply:
@gchanan do you want to take another look at this